tutorials/6_comp_invars.md

```diff
@@ -308,15 +308,32 @@ indeed adequate.
 However, we still have yet to prove the compositionality of our new candidate.
 We can automate this check by instrumenting one final model. We do this by
 encoding `Initialization`, `Local Inductiveness` and `Non-interference` as
-program assertions. While most of the code is fairly straight forward, we do
-make use of two modeling tricks.
-
-  1. When checking `Initialization` and `Non-interference` we use three
-     non-deterministic flag variables to model overlap between the two
-     neighbourhoods.
-  2. To simulate an external process, we compute two neighbourhoods for the same
+program assertions.
+
+The check for `Local Inductiveness` follows directly from the definition. The
+check of `Initialization` is somewhat more nuanced. The challenge here is that
+after initialization, our view is of the neighbourhood which was acted on by the
+constructor. In reality, all other representatives are zero initialized, and may
```

Comment:
I'm still struggling with understanding the two challenges properly. Maybe a more concrete example would help. Or maybe there is a naive approach for solving the challenges. Then I could understand the "tricks" as an optimization.

Reply:
Ah I think I see the confusion. Yes, these "tricks" can be understood as an optimization, if I am to understand your point correctly. Without this "trick" there's 2^n possibilities to consider (where n is the size of the neighbourhood).

I will add a remark about the naive approach. Hopefully that will clarify what's going on.

Reply:
That definitely helps. :) I added some more questions.